lib/facter/udevadm.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# Fact: udevadm
+#
+# Purpose:
+#   Provide information about the udevadm utilty.
+#
+# Resolution:
+#   Check if the kernel is linux.
+#
+# Caveats:
+#
+Facter.add(:udevadm) do
+  confine kernel: :linux
+
+  setcode do
+    path = Facter::Core::Execution.which('udevadm')
+    return nil if path.nil?
```

Comment:
I don't think you can use `return` in blocks. At least, pretty sure I had problems with that in the past.

Reply:
Yes, that was a silly mistake. I will change it to break.

Reply:
`return` has been replaced with `nil`.